### PR TITLE
added --export-as-hardlink option

### PR DESCRIPTION
**Pull request by britiscurious**
my very first step into the python world :-D

I added the --export-as-hardlink option and corresponding implementation.

### CODE REVIEW

File: osxphotos/utils.py [hunk 1]
```diff
@@ -118,6 +118,32 @@ def _dd_to_dms(dd):
...
+        dest: destination path as string
+        Raises exception if linking fails or either path is None """
+
+    if src is None or dest is None:
+        raise ValueError("src and dest must not be None", src, dest)
+
+    if not os.path.isfile(src):
+        raise FileNotFoundError("src file does not appear to exist", src)
+
+    command = ["ln", src, dest]
```
Comment (RhetTbull) [inline osxphotos/utils.py:133]:
```suggestion
```

File: osxphotos/utils.py [hunk 2]
```diff
@@ -118,6 +118,32 @@ def _dd_to_dms(dd):
...
+
+    # if error on copy, subprocess will raise CalledProcessError
+    try:
+        result = subprocess.run(command, check=True, stderr=subprocess.PIPE)
```
Comment (RhetTbull) [inline osxphotos/utils.py:137]:
```suggestion
        os.link(src, dest)
```

File: osxphotos/utils.py [hunk 3]
```diff
@@ -118,6 +118,32 @@ def _dd_to_dms(dd):
...
+    except subprocess.CalledProcessError as e:
```
Comment (RhetTbull) [inline osxphotos/utils.py:138]:
```suggestion
    except Exception as e:
```

File: osxphotos/utils.py [hunk 4]
```diff
@@ -118,6 +118,32 @@ def _dd_to_dms(dd):
...
+        logging.critical(
+            f"ln returned error: {e.returncode} {e.stderr.decode(sys.getfilesystemencoding()).rstrip()}"
+        )
+        raise e
+
+    return result.returncode
```
Comment (RhetTbull) [inline osxphotos/utils.py:144]:
```suggestion
```

**Comment by RhetTbull** [comment]
Also, please update `osxphotos/_version.py` to bump the minor version number.

https://github.com/RhetTbull/osxphotos/blob/7444b6d173918a3ad2a07aefce5ecf054786c787/osxphotos/_version.py#L3

e.g. change 0.28.13 to 0.28.14

**Comment by britiscurious** [comment]
:-) Not only python is new to me. Same for Github contribution and the related workflow.

**Comment by RhetTbull** [comment]
Great job on your very first python project and first GitHub contribution!  Thanks for contributing!

**Pull request merged**